### PR TITLE
[phi] Transfer linear_interp_v2 yaml to phi

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
New features

### PR changes
OPs

### Describe
- Transfer linear_interp_v2 yaml to phi
- support `optional<vector<Tensor>>` data structure 
- fix linear_interp_v2 testcase
- fix some typo error

### next PR todo

- rename linear_interp_v2 to linear_interp
- transfer another 4 interpolate ops to phi

### CODE REVIEW

**Review by dingjiaweiww — APPROVED** [review]

**Review by Xreki — APPROVED** [review]
LGTM for the setting of atol

File: python/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -590,8 +590,24 @@ def _is_list_or_turple_(data):
...
+        ])
+        eager_args.append(attrs['scale'] if 'scale' in attrs else [])
+        eager_args.extend([
+            attrs['interp_method'], attrs['align_corners'], attrs['align_mode']
+        ])
+
         if resample_type == "linear":
-            out = _C_ops.linear_interp_v2(x, *dy_attr)
+            if in_dygraph_mode():
+                out = _C_ops.final_state_linear_interp_v2(*eager_args)
```
Comment (zyfncg) [inline python/paddle/nn/functional/common.py:608]: 这里写成逐个参数传入的形式可能会更好一些

**Review by zyfncg — APPROVED** [review]

**Review by YuanRisheng — APPROVED** [review]

File: python/paddle/nn/functional/common.py (hunk 1 above)
Comment (jiahy0825) [reply python/paddle/nn/functional/common.py:608]: 下个PR改

File: paddle/phi/api/yaml/legacy_api.yaml [hunk 2]
```diff
@@ -1409,6 +1409,17 @@
   kernel :
     func : less_than
 
+- api : linear_interp_v2
```
Comment (chenwhql) [inline paddle/phi/api/yaml/legacy_api.yaml:1412]: api名字不可以带v2，在sig.cc里面映射一下，再向外暴露

Comment (jiahy0825) [reply paddle/phi/api/yaml/legacy_api.yaml:1412]: 下个PR迁移时，五个算子统一改

**Review by chenwhql — APPROVED** [review]

**Pull request merged**